routes/s3handler.js

```diff
@@ -0,0 +1,239 @@
+import express from 'express'
+import CryptoJS from 'crypto-js'
+import bodyParser from 'body-parser'
+import getObjectURLFromS3 from '../src/utils/s3'
+import faceUtils from '../src/utils/face++'
+
+const clientSecretKey = process.env.AWS_SECRET_ACCESS_KEY
+const expectedMinSize = 0
+const expectedMaxSize = 5000000
+const expectedBucket = 'uploadedphotostomatch'
+const expectedHostname = 'uploadedphotostomatch.s3.amazonaws.com'
+
+const router = express.Router()
+
+// Signs any requests.  Delegate to a more specific signer based on type of request.
+function signRequest(req, res) {
+    if (req.body.headers) {
+        signRestRequest(req, res);
+    }
+    else {
+        signPolicy(req, res);
+    }
+}
+
+// Signs multipart (chunked) requests.
+function signRestRequest(req, res) {
+    var version = req.query.v4 ? 4 : 2,
+        stringToSign = req.body.headers,
+        signature = version === 4 ? signV4RestRequest(stringToSign) : signV2RestRequest(stringToSign);
```

Comment:
Might want to name these [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) with constants.

Reply:
Is a magic number here just  `const FOUR = 4 and const TWO = 2`?

Reply:
On second thought, why not make this simpler by just checking for `req.query.v4`? No need to have another variable.